plugin/evm/status.go

```diff
@@ -0,0 +1,81 @@
+// (c) 2019-2020, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package evm
+
+import (
+	"errors"
+)
+
+var (
+	errUnknownStatus = errors.New("unknown status")
+)
+
+// Status ...
+type Status uint32
+
+// List of possible status values
+// [Unknown] Zero value, means the status is not known
+// [Dropped] means the transaction was in the mempool, but was dropped because it failed verification
+// [Processing] means the transaction is in the mempool
+// [Accepted] means the transaction was accepted
+const (
+	Unknown Status = iota
+	Dropped
+	Processing
+	Accepted
+)
+
+// MarshalJSON ...
+func (s Status) MarshalJSON() ([]byte, error) {
+	if err := s.Valid(); err != nil {
+		return nil, err
+	}
+	return []byte("\"" + s.String() + "\""), nil
+}
+
+// UnmarshalJSON ...
+func (s *Status) UnmarshalJSON(b []byte) error {
+	str := string(b)
+	if str == "null" {
+		return nil
+	}
+	switch str {
+	case "\"Unknown\"":
```

Comment:
nit: `` `"Unknown"` `` ... may also want to make a const

Reply:
We can probably use this cleanup in a couple places.

Reply:
Going to leave this as is for now, since if we add this nit, it should probably be done in AvalancheGo and imported to coreth.